src/main/scala/nodes/learning/KernelRidgeRegression.scala

```diff
@@ -0,0 +1,343 @@
+package nodes.learning
+
+import workflow.LabelEstimator
+
+import breeze.linalg._
+import breeze.math._
+import breeze.numerics._
+import org.apache.spark.rdd.RDD
+import org.apache.spark.broadcast.Broadcast
+import org.apache.spark.SparkConf
+import org.apache.spark.SparkContext
+import org.apache.spark.SparkContext._
+import utils._
+
+
+import edu.berkeley.cs.amplab.mlmatrix.RowPartitionedMatrix
+import edu.berkeley.cs.amplab.mlmatrix.util.{Utils => MLMatrixUtils}
+
+import scala.util.Random
+
+class KernelRidgeRegression(
+  kernelBlockGenerator: KernelGenerator,
```

Comment:
Partly - but not completely. There are a couple of things which aren't different IMHO from your description

- In theory don't need to do any estimation for the KernelGenerator. The Generator is just a transformer which takes in a RDD[DenseVector] (i.e. training data) in its constructor.  I don't know if that is elegant to do in Keystone
- The LinearSystemSolver is different from LeastSquares -- The former solves `Ax = b` while the latter solves `|| Ax - b||_2`.  We definitely want to use the former after applying the kernel.

- Then your pipeline can be something like 
```
val trainData = preProcessing(inputData) 
val kernel = new RBFKernelGenerator(trainData)
val predictorPipeline = preProcessing andThen kernel andThen new LinearSystemSolver(blockSize, lambda)
```
- So the only weird thing about the code above is that the RBFGenerator can't be chained using `andThen`. I guess if you make it an `Estimator` then it might be possible ?